x-pack/metricbeat/cmd/root.go

```diff
@@ -71,10 +72,27 @@ func defaultProcessors() []mapstr.M {
 	//   - add_cloud_metadata: ~
 	//   - add_docker_metadata: ~
 	//   - add_kubernetes_metadata: ~
-	return []mapstr.M{
-		{"add_host_metadata": nil},
-		{"add_cloud_metadata": nil},
-		{"add_docker_metadata": nil},
-		{"add_kubernetes_metadata": nil},
+
+	// Setting environmental variable NETINFO:false in Elastic Agent pod will disable the netinfo.enabled option of add_host_metadata processor.
+	// This will result to events not being enhanced with host.ip and host.mac
+	// Related to https://github.com/elastic/integrations/issues/6674
+	valueNETINFO, status := os.LookupEnv("NETINFO")
+
+	if valueNETINFO == "false" && status == true {
+		return []mapstr.M{
+			{"add_host_metadata": mapstr.M{
+				"netinfo.enabled": "false",
```

Comment:
Why to enable/disable this at this level and not inside the processor's implementation?
Wouldn't that be more generic without the need to configure every single Beat? And that would cover @ruflin 's concern at https://github.com/elastic/beats/pull/36506#issuecomment-1707786366, right?

Reply:
👍 to do that inside the processor implementation. I think in the implementation we could even automatically check if this is running inside k8s or inside a container and either disable netinfo completely or filter out link-local addresses. See also https://github.com/elastic/integrations/issues/6674#issuecomment-1706768981.

Reply:
Adding the fix only to add-host-metadata processor (https://github.com/elastic/beats/pull/36506/files#diff-03b979565735e6707425275586e3215f44192f46e1ac3b456516d93c653eba13R38) seems that works only for metricbeat. Somehow seems we override it for filestream? Will keep looking